packages/core/src/render3/instructions/styling.ts

```diff
@@ -304,322 +159,473 @@ export function ɵɵstyleMap(styles: {[styleName: string]: any} | NO_CHANGE | nu
+ * @param lView `LView` contains the values associated with other styling binding at this `TNode`.
+ * @param renderer Renderer to use if any updates.
+ * @param prop Either style property name or a class name.
+ * @param value Either style vale for `prop` or `true`/`false` if `prop` is class.
+ * @param isClassBased `true` if `class` (`false` if `style`)
+ * @param bindingIndex Binding index of the binding.
  */
-function stylingApply(): void {
-  const lView = getLView();
-  const tView = lView[TVIEW];
-  const elementIndex = getSelectedIndex();
-  const tNode = getTNode(elementIndex, lView);
-  const native = getNativeByTNode(tNode, lView) as RElement;
-  const directiveIndex = getActiveDirectiveId();
-  const renderer = getRenderer(tNode, lView);
-  const sanitizer = getCurrentStyleSanitizer();
-  const classesContext = isStylingContext(tNode.classes) ? tNode.classes as TStylingContext : null;
-  const stylesContext = isStylingContext(tNode.styles) ? tNode.styles as TStylingContext : null;
-  flushStyling(
-      renderer, lView, tNode, classesContext, stylesContext, native, directiveIndex, sanitizer,
-      tView.firstUpdatePass);
-  resetCurrentStyleSanitizer();
-}
-
-function getRenderer(tNode: TNode, lView: LView) {
-  return tNode.type === TNodeType.Element ? lView[RENDERER] : null;
+function updateStyling(
+    tView: TView, tNode: TNode, lView: LView, renderer: Renderer3, prop: string,
+    value: string | undefined | null | boolean, isClassBased: boolean, bindingIndex: number) {
+  if (tNode.type !== TNodeType.Element) {
+    // It is possible to have styling on non-elements (such as ng-container).
+    // This is rare, but it does happen. In such a case, just ignore the binding.
+    return;
+  }
+  const tData = tView.data;
+  const tRange = tData[bindingIndex + 1] as TStylingRange;
+  const higherPriorityValue = getTStylingRangeNextDuplicate(tRange) ?
+      findStylingValue(tData, null, lView, prop, getTStylingRangeNext(tRange), isClassBased) :
+      undefined;
+  if (!isStylingValuePresent(higherPriorityValue)) {
+    // We don't have a next duplicate, or we did not find a duplicate value.
+    if (!isStylingValuePresent(value)) {
+      // We should delete current value or restore to lower priority value.
+      if (getTStylingRangePrevDuplicate(tRange)) {
+        // We have a possible prev duplicate, let's retrieve it.
+        value =
+            findStylingValue(tData, tNode, lView, prop, getTStylingRangePrev(tRange), isClassBased);
+      }
+    }
+    const rNode = getNativeByIndex(getSelectedIndex(), lView) as RElement;
+    applyStyling(renderer, isClassBased, rNode, prop, value);
+  }
 }
 
 /**
- * Searches and assigns provided all static style/class entries (found in the `attrs` value)
- * and registers them in their respective styling contexts.
+ * Search for styling value with higher priority which is overwriting current value.
+ *
+ * When value is being applied at a location related values need to be consulted.
```

Comment:
Nit: the comma is necessary here because it sounds like "location-related values"

```suggestion
 * When value is being applied at a location, related values need to be consulted.
```

Reply:
fixed